src/app/+admin/admin-registries/metadata-schema/metadata-field-form/metadata-field-form.component.ts

```diff
@@ -0,0 +1,200 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { MetadataSchema } from '../../../../core/metadata/metadataschema.model';
+import {
+  DynamicFormControlModel,
+  DynamicFormGroupModel,
+  DynamicFormLayout,
+  DynamicInputModel
+} from '@ng-dynamic-forms/core';
+import { FormGroup } from '@angular/forms';
+import { RegistryService } from '../../../../core/registry/registry.service';
+import { FormBuilderService } from '../../../../shared/form/builder/form-builder.service';
+import { Observable } from 'rxjs/internal/Observable';
+import { MetadataField } from '../../../../core/metadata/metadatafield.model';
+import { take } from 'rxjs/operators';
+import { TranslateService } from '@ngx-translate/core';
+import { combineLatest } from 'rxjs/internal/observable/combineLatest';
+
+@Component({
+  selector: 'ds-metadata-field-form',
+  templateUrl: './metadata-field-form.component.html'
+})
+/**
+ * A form used for creating and editing metadata fields
+ */
+export class MetadataFieldFormComponent implements OnInit, OnDestroy {
+
+  /**
+   * A unique id used for ds-form
+   */
+  formId = 'metadata-field-form';
+
+  /**
+   * The prefix for all messages related to this form
+   */
+  messagePrefix = 'admin.registries.schema.form';
+
+  /**
+   * The metadata schema this field is attached to
+   */
+  @Input() metadataSchema: MetadataSchema;
+
+  /**
+   * A dynamic input model for the element field
+   */
+  element: DynamicInputModel;
+
+  /**
+   * A dynamic input model for the qualifier field
+   */
+  qualifier: DynamicInputModel;
+
+  /**
+   * A dynamic input model for the scopeNote field
+   */
+  scopeNote: DynamicInputModel;
+
+  /**
+   * A list of all dynamic input models
+   */
+  formModel: DynamicFormControlModel[];
+
+  /**
+   * Layout used for structuring the form inputs
+   */
+  formLayout: DynamicFormLayout = {
+    element: {
+      grid: {
+        host: 'col col-sm-6 d-inline-block'
+      }
+    },
+    qualifier: {
+      grid: {
+        host: 'col col-sm-6 d-inline-block'
+      }
+    },
+    scopeNote: {
+      grid: {
+        host: 'col col-sm-12 d-inline-block'
+      }
+    }
+  };
+
+  /**
+   * A FormGroup that combines all inputs
+   */
+  formGroup: FormGroup;
+
+  /**
+   * An EventEmitter that's fired whenever the form is being submitted
+   */
+  @Output() submitForm: EventEmitter<any> = new EventEmitter();
+
+  constructor(public registryService: RegistryService,
+              private formBuilderService: FormBuilderService,
+              private translateService: TranslateService) {
+  }
+
+  ngOnInit() {
```

Comment:
Like you've done with `onCancel` and `onSubmit`, it'd be good to add a basic description of what `ngOnInit()` is doing here, as it's rather large.

Reply:
Added